synapse/rest/client/transactions.py

```diff
@@ -21,10 +21,10 @@
 
 from twisted.internet.defer import Deferred
 from twisted.python.failure import Failure
-from twisted.web.server import Request
+from twisted.web.iweb import IRequest
```

Comment:
Overall seems cleaner. I think I started doing that because I wanted to have `fetch_or_execute_request` pass the requester and the request automatically, and wanted the Requester parameter to be generic 🤔

Reply:
If we don't need any of the features of `twisted.web.server.Request` than that's fine. 👍